src/vsphere_cpi/spec/integration/human_readable_names_spec.rb

```diff
@@ -0,0 +1,134 @@
+require 'integration/spec_helper'
+
+describe 'enable human readable names' do
+  let(:vm_type) do
+    {
+        'ram' => 512,
+        'disk' => 2048,
+        'cpu' => 1,
+    }
+  end
+
+  let(:network_spec) do
+    {
+      'static' => {
+        'ip' => "169.254.#{rand(1..254)}.#{rand(4..254)}",
+        'netmask' => '255.255.254.0',
+        'cloud_properties' => {'name' => @vlan},
+        'default' => ['dns', 'gateway'],
+        'dns' => ['169.254.1.2'],
+        'gateway' => '169.254.1.3'
+      }
+    }
+  end
+
+  let(:human_readable_name_cpi) do
+    options = cpi_options({'enable_human_readable_name' => true})
+    VSphereCloud::Cloud.new(options)
+  end
+
+  let(:machine_name_cpi) do
+    options = cpi_options({'enable_human_readable_name' => false})
+    VSphereCloud::Cloud.new(options)
+  end
+
+  context 'When enable human readable names is set to false' do
+    let(:environment){
+      {'bosh' => { 'groups' => ['fake-director-name', 'fake-deployment-name', 'fake-instance-group-name'] }
+      }
+    }
+    it 'create vm with UUID based name' do
+      begin
+        test_vm_id = machine_name_cpi.create_vm(
+          'agent-007',
+          @stemcell_id,
+          vm_type,
+          network_spec,
+          [],
+          environment
+        )
+        expect(test_vm_id).to_not be_nil
+        expect(test_vm_id.size).to eq 39
+        expect(test_vm_id).to start_with('vm')
```

Comment:
We should probably use a regular expression match here instead of `start_with('vm')`. In fact we can probably replace all three lines with a single regular expression match.

We can probably even mock out the `SecureRandom.uuid` method here for an exact match.